business/istio_validations.go

```diff
@@ -21,177 +21,106 @@ type ObjectChecker interface {
 	Check() models.IstioValidations
 }
 
-// GetServiceValidations returns an IstioValidations object with all the checks found when running
+// GetValidations returns an IstioValidations object with all the checks found when running
 // all the enabled checkers.
-func (in *IstioValidationsService) GetServiceValidations(namespace, service string) (models.IstioValidations, error) {
+func (in *IstioValidationsService) GetValidations(namespace, service string) (models.IstioValidations, error) {
 	var err error
 	promtimer := internalmetrics.GetGoFunctionMetric("business", "IstioValidationsService", "GetServiceValidations")
 	defer promtimer.ObserveNow(&err)
 
-	// Ensure the service exists
-	if _, err := in.k8s.GetService(namespace, service); err != nil {
-		return nil, err
+	// Ensure the service or namespace exists.. do we need to block with this?
+	if service != "" {
+		if _, err := in.k8s.GetService(namespace, service); err != nil {
+			return nil, err
+		}
+	} else {
+		if _, err := in.k8s.GetNamespace(namespace); err != nil {
+			return nil, err
+		}
 	}
 
-	// Get Gateways and ServiceEntries to validate VirtualServices
 	wg := sync.WaitGroup{}
-	errChan := make(chan error, 64)
+	errChan := make(chan error, 1)
 
-	istioDetails := kubernetes.IstioDetails{}
-	vs := make([]kubernetes.IstioObject, 0)
-	drs := make([]kubernetes.IstioObject, 0)
-	gws := make([]kubernetes.IstioObject, 0)
+	var istioDetails kubernetes.IstioDetails
 	var services []v1.Service
 	var workloads models.WorkloadList
+	var gatewaysPerNamespace [][]kubernetes.IstioObject
+
+	wg.Add(4) // We need to add these here to make sure we don't execute wg.Wait() before scheduler has started goroutines
+
+	// NoServiceChecker is not necessary if we target a single service - those components with validation errors won't show up in the query
+	go in.serviceFetcher(&services, namespace, service, errChan, &wg)
```

Comment:
I wonder whether unified the name "serviceFetcher -> fetchServices" and also in next line "detailsFetcher -> fetchIstioDetails" and "getGatewaysPerNamespace -> fetchGatewaysPerNamespace" will help to future maintainers. 
I know, it's pure naming comment.

Reply:
I'll rename.